quarkchain/evm/vm.py

```diff
@@ -683,7 +683,16 @@ def vm_execute(ext, msg, code):
                 ingas = compustate.gas
                 if ext.post_anti_dos_hardfork():
                     ingas = all_but_1n(ingas, opcodes.CALL_CHILD_LIMIT_DENOM)
-                create_msg = Message(msg.to, b"", value, ingas, cd, msg.depth + 1)
+                create_msg = Message(
+                    msg.to,
+                    b"",
+                    value,
+                    ingas,
+                    cd,
+                    msg.depth + 1,
+                    from_full_shard_id=msg.from_full_shard_id,
```

Comment:
from_full_shard_id is not used in create_contract.

it should also be msg.to_full_shard_id which is the shard id of the originating contract.

Reply:
done